Networking/Networking/Remote/SiteVisitStatsRemote.swift

```diff
@@ -48,3 +53,37 @@ private extension SiteVisitStatsRemote {
         static let statFields: String = "stat_fields"
     }
 }
+
+/// An error that occurs at site visit stats networking layer.
+/// API documentation of possible errors:
+/// https://developer.wordpress.com/docs/api/1.1/get/sites/%24site/stats/
+///
+public enum SiteVisitStatsRemoteError: Error {
+    case statsModuleDisabled
+    case unknown
+
+    private enum ErrorIdentifiers {
+        static let invalidBlog: String = "invalid_blog"
+    }
+
+    private enum ErrorMessages {
+        static let statsModuleDisabled: String = "This blog does not have the Stats module enabled"
+    }
+
+    init(error: Error) {
+        guard let dotcomError = error as? DotcomError else {
+            self = .unknown
+            return
+        }
+        switch dotcomError {
+        case .unknown(let code, let message):
+            if code == ErrorIdentifiers.invalidBlog && message == ErrorMessages.statsModuleDisabled {
```

Comment:
This seems prone to break, if the error message changes. I am not sure if it's even possible, or if it would be performant enough, or if it would even make sense, but what about relying on the Natural Language Framework to try to identify the likelihood of the error being the module disabled?

Reply:
Oops, I just saw this other comment: https://github.com/woocommerce/woocommerce-ios/issues/1070#issuecomment-511566383

Please ignore me!

Reply:
yes I'd much prefer if the check does not rely on a natural language message comparison. I hope in the future, the error code from the API is unique for each scenario so that the client side can handle each case in a less error-prone way